### PR TITLE
ParityShell::open `Return result`

### LINKED ISSUE #8369: parity::open should return error if it fails
discussion: https://github.com/paritytech/parity/pull/8366#discussion_r180733188

### PR DESCRIPTION
**Pull request by niklasad1**
Attempt to close #8369

### CODE REVIEW

File: parity/url.rs [hunk 1]
```diff
@@ -16,33 +16,70 @@
...
+	use winapi::shared::minwindef::INT;
+
+	const WINDOWS_SHELL_EXECUTE_SUCCESS: i32 = 32;
 
-	unsafe {
+	let h_instance = unsafe {
 		ShellExecuteA(ptr::null_mut(),
-			CString::new("open").unwrap().as_ptr(),
-			CString::new(url.to_owned().replace("\n", "%0A")).unwrap().as_ptr(),
+			CString::new("open")?.as_ptr(),
```
Comment (tomaka) [inline parity/url.rs]: Isn't `unwrap` "better" for `CString::new("open")`? It logically shouldn't fail.

File: parity/url.rs [hunk 2]
```diff
@@ -16,33 +16,70 @@
 
 //! Cross-platform open url in default browser
 
+use std;
+
+#[allow(unused)]
+pub enum Error {
+	ProcessError(std::io::Error),
+	FfiNull(std::ffi::NulError),
+	WindowsShellExecute,
```
Comment (tomaka) [inline parity/url.rs]: I'd suggest putting the return value of `ShellExecuteA` (casted as a `c_int`) as parameter, for easier debugging if that actually happens.

File: parity/url.rs (hunk 1 above)
Comment (niklasad1) [reply parity/url.rs]: Yeah, you are right it should not fail unless the `&str` contains `0`!

`let ff = CString::new("\u{0}").unwrap();` will fail!

**Comment by 5chdn** [comment]
@tomaka can I dismiss your review or do you want to look at the changes again?

**Pull request merged**